### PR TITLE
implimenting isfinite function

### PR DESCRIPTION
**Pull request by ndurumo254**

### CODE REVIEW

File: ivy/functional/frontends/jax/numpy/logic.py [hunk 1]
```diff
@@ -147,3 +146,7 @@ def invert(x, /):
 @to_ivy_arrays_and_back
 def isinf(x, /):
     return ivy.isinf(x)
+
+
```
Comment (nassimberrada) [inline ivy/functional/frontends/jax/numpy/logic.py:150]: You may need to add the @to_ivy_arrays_and_back decorator

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py [hunk 2]
```diff
@@ -703,14 +703,38 @@ def test_jax_numpy_invert(
...
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        x=x[0],
+    )
+
+
```
Comment (nassimberrada) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py:728]: The hypothesis part of the tests is missing, you can take a look at other functions tests (or even the tests for isfinite as implemented in torch for e.g) to get a better idea of the general structure, it's the part that starts with the @handle_frontend_test decorator

**Review by nassimberrada — CHANGES_REQUESTED** [review]

**Comment by nassimberrada** [comment]
Hi @ndurumo254, thanks for your contribution ! You're in the right direction but some things should be changed, i've made some comments above could you try addressing those ?

Also, i see you've made changes to the numpy backend function as well as the tests for other frontend functions, those are not necessary to implement this particular function so maybe you could revert them back, just to make your PR's purpose clear, of course you are free to make unrelated corrections other PRs. 

Finally, could you reserve this function for yourself by creating an issue, commenting it in the corresponding To-Do list and linking it to this PR ? You can find more details about this [here](https://lets-unify.ai/ivy/contributing/the_basics.html#the-basics).

Let me know when you feel your PR is ready and you've made sure the tests pass so i can review it again, thanks !

**Comment by ndurumo254** [comment]
> Hi @ndurumo254, thanks for your contribution ! You're in the right direction but some things should be changed, i've made some comments above could you try addressing those ?
> 
> Also, i see you've made changes to the numpy backend function as well as the tests for other frontend functions, those are not necessary to implement this particular function so maybe you could revert them back, just to make your PR's purpose clear, of course you are free to make unrelated corrections other PRs.
> 
> Finally, could you reserve this function for yourself by creating an issue, commenting it in the corresponding To-Do list and linking it to this PR ? You can find more details about this [here](https://lets-unify.ai/ivy/contributing/the_basics.html#the-basics).
> 
> Let me know when you feel your PR is ready and you've made sure the tests pass so i can review it again, thanks !

thanks for your response @nassimberrada . i have made changes to the tests. you can review it.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py [hunk 3]
```diff
@@ -724,3 +724,34 @@ def test_jax_numpy_isinf(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=with_out,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        input=input[0],
```
Comment (nassimberrada) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py:756]: This should be `x=x[0]` given that your frontend implementation uses x as variable name

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py [hunk 4]
```diff
@@ -724,3 +724,34 @@ def test_jax_numpy_isinf(
...
+        dtype_and_input,
+        as_variable,
+        with_out,
+        num_positional_args,
+        native_array,
+        on_device,
+        fn_tree,
+        frontend,
+):
+    input_dtype, input = dtype_and_input
```
Comment (nassimberrada) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_logic.py:746]: Likewise, this should probably be written as `input_dtype, x = dtype_and_x`

**Review by nassimberrada — CHANGES_REQUESTED** [review]

**Comment by nassimberrada** [comment]
Hi @ndurumo254, thanks for making these changes. I've made some more comments to address some issues with the testing. 

Also i see that the unrelated changes (to the numpy backend function, and to other frontend functions) persist, could you revert those ? Even if your changes are correct it would be best to separate them from this PR so that the purpose of each PR remains clear when debugging.

I suggest you take a look at the [ivy frontend tests](https://lets-unify.ai/ivy/deep_dive/ivy_frontends_tests.html) section of the docs to guide you in implementing the tests for your frontend function. Thanks !

**Comment by ndurumo254** [comment]
hello @nassimberrada ,hope this finds you well. in the effoort of reverting the unrelated changes i get rest --hard to previous commit and thereafter the behavior of the tests has changed. was i doing the right thing?

**Comment by nassimberrada** [comment]
Hi @ndurumo254, it seems this created some issues in your PR, could you try to sync your fork with the master branch by going to your fork, clicking on "Sync fork" then "Update Branch", as shown in the following screenshot ?

image

**Comment by nassimberrada** [comment]
Hi, i think some unintended changed were made as you tried to revert your commits, one of our team members fixed those so your PR is back to normal, feel free to try making the changes i suggested again, thanks !

**Pull request closed without merging**